redis/src/push_manager.rs

```diff
@@ -0,0 +1,61 @@
+use crate::{PushKind, RedisResult, Value};
+use arc_swap::ArcSwap;
+use std::sync::Arc;
+
+/// Holds information about received Push data
+#[derive(Debug, Clone)]
+pub struct PushInfo {
+    /// Push Kind
+    pub kind: PushKind,
+    /// Data from push message
+    pub data: Vec<Value>,
+
+    /// Connection address to distinguish connections
+    pub con_addr: Arc<String>,
+}
+
+/// Manages Push messages for both tokio and std channels
+#[derive(Clone, Default)]
+pub struct PushManager {
+    sender: Arc<ArcSwap<Option<tokio::sync::mpsc::UnboundedSender<PushInfo>>>>,
+}
+impl PushManager {
+    /// It checks if value's type is Push
+    /// then invokes `try_send_raw` method
+    pub(crate) fn try_send(&self, value: &RedisResult<Value>, con_addr: &Arc<String>) {
+        if let Ok(value) = &value {
+            self.try_send_raw(value, con_addr);
+        }
+    }
+
+    /// It checks if value's type is Push and there is a provided sender
+    /// then creates PushInfo and invokes `send` method of sender
+    pub(crate) fn try_send_raw(&self, value: &Value, con_addr: &Arc<String>) {
+        if let Value::Push { kind, data } = value {
```

Comment:
@altanozlu did you see this comment?

Reply:
Hi! mpxc still needs a function for &Value, because it's `Result<Value, E>` not a RedisResult. Also I think this function should stay as a no-op on non-push values. Your function always loads the value(probably cheap). So I'm not sure if we should go further with it.